tools/integration-tests/tester/tests/testutil.go

```diff
@@ -703,19 +780,36 @@ func AwaitTransactionInclusionState(peers []*framework.Peer, transactionIDs map[
 					if expInclState.Solid != nil && *expInclState.Solid != metadata.Solid {
 						continue
 					}
-					atomic.AddInt32(&counter, -1)
+					unmatchedMu.Lock()
+					delete(unmatched[p.ID()], txID)
+					if len(unmatched[p.ID()]) == 0 {
+						delete(unmatched, p.ID())
+					}
+					unmatchedMu.Unlock()
 				}
 			}(p)
 		}
+
 		wg.Wait()
-		if counter == 0 {
+		if len(unmatched) == 0 {
 			// everything available
 			return nil
 		}
+
+		i++
 	}
 	return ErrTransactionStateNotSameInTime
 }
 
+// AwaitCheck waits until the given peers are in synced.
+func AwaitCheck(f func(t *testing.T, peers []*framework.Peer), maxAwait time.Duration) error {
+	s := time.Now()
+	for ; time.Since(s) < maxAwait; time.Sleep(retryInterval) {
+
```

Comment:
isn't there some polling logic missing here?

Reply:
I think I messed things up during merging files. :stuck_out_tongue: 
We don't need this function.